### ISSUE TITLE
Confirm that use of BLAST's `-max_target_seqs` is intentional

### ISSUE THREAD

**Issue by armish** [dnasko/CASC#7]
Hi there,

This is a semi-automated message from a fellow bioinformatician. Through a GitHub search, I found that the following source files make use of BLAST's `-max_target_seqs` parameter: 

- [src/para_blastx.pl](https://github.com/dnasko/CASC/blob/fed9dd99f321fbee12826704ecad09231f9565c8/src/para_blastx.pl)

Based on the recently published report, [Misunderstood parameter of NCBI BLAST impacts the correctness of bioinformatics workflows](https://academic.oup.com/bioinformatics/advance-article-abstract/doi/10.1093/bioinformatics/bty833/5106166?redirectedFrom=fulltext), there is a strong chance that this parameter is misused in your repository.

If the use of this parameter was intentional, please feel free to ignore and close this issue but I would highly recommend to add a comment to your source code to notify others about this use case. If this is a duplicate issue, please accept my apologies for the redundancy as this simple automation is not smart enough to identify such issues.

Thank you!
-- Arman ([armish/blast-patrol](https://github.com/armish/blast-patrol))

**Comment by dnasko** [comment]
@armish you should be searching for instances where `-max_target_seqs` is set to 1 (or a similarly small number). My code uses a value exceeding the number of sequences in the database (`-max_target_seqs 10000000`) so it will return what I expect (i.e. all significant hits). Thus, my implementation is correct.

As a matter of fact your bot should be checking for all instances of `blast` being used with `-outfmt 6` because even when `-max_target_seqs` **isn't** set by the user, in this output format, it's set to a default of 500, which may not be a large enough value to guarantee you return the best hit. This is something they don't mention in the Letter to the Editor (and that I'll have to bring up tomorrow in lab meeting, because I work with the authors!).

**Comment by armish** [comment]
gotcha @dnasko — sorry about this false positive hit. The automated system works in the most simple way and things could have been much better as you mentioned. Unfortunately, after doing a few manual searches, I realized that people were also misusing this parameter and were trying to get, say top 10, in a majority of the repositories I saw. Hence, it was hard to decide what would be a good cut-off threshold so I ended up not defining one.

Your improvement suggestions do make sense but unfortunately we are also limited by GitHub's code search API limits. For example, finding all `blast` calls with `outfmt` parameter would be a non-trivial task.

I appreciate you taking the time to respond to this issue. 

All the best,
-- Arman
